lib/vector-core/src/event/metric/tags.rs

```diff
@@ -5,12 +5,13 @@ use std::hash::{Hash, Hasher};
 use std::{cmp::Ordering, mem};
 
 use indexmap::IndexSet;
-use serde::{Deserialize, Deserializer, Serialize, Serializer};
+use serde::{ser::SerializeSeq, Deserialize, Deserializer, Serialize, Serializer};
 use vector_common::byte_size_of::ByteSizeOf;
 use vector_config::{configurable_component, Configurable};
 
 /// A single tag value, either a bare tag or a value.
 #[derive(Clone, Configurable, Debug, Deserialize, Eq, Hash, PartialEq, Serialize)]
+#[serde(untagged)]
```

Comment:
Hmm, is my intuition correct here that, with the change to be untagged... a null value for a field of type `TagValue` will get deserialized as `TagValue::Bare`?

Reply:
That is the intention, yes.